internal/policy/policy.go

```diff
@@ -765,3 +770,54 @@ func (s *State) findDelegationEntry(roleName string) (*tuf.Delegation, error) {
 		}
 	}
 }
+
+func ListRules(ctx context.Context, repo *git.Repository) ([]*DelegationDepth, error) {
+	state, err := LoadCurrentState(ctx, repo)
+	if err != nil {
+		return nil, err
+	}
+
+	topLevelTargetsMetadata, err := state.GetTargetsMetadata(TargetsRoleName)
+	if err != nil {
+		return nil, err
+	}
+
+	delegationTargetsMetadata := map[string]*tuf.TargetsMetadata{}
+	for name, env := range state.DelegationEnvelopes {
+		targetsMetadata := &tuf.TargetsMetadata{}
+
+		envBytes, err := env.DecodeB64Payload()
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal(envBytes, targetsMetadata); err != nil {
+			return nil, err
+		}
+		delegationTargetsMetadata[name] = targetsMetadata
+	}
+
+	delegationsToSearch := []*DelegationDepth{}
+	allDelegations := []*DelegationDepth{}
+
+	for _, topLevelDelegation := range topLevelTargetsMetadata.Delegations.Roles {
+		delegationsToSearch = append(delegationsToSearch, &DelegationDepth{Delegation: topLevelDelegation, Depth: 0})
+		allDelegations = append(allDelegations, &DelegationDepth{Delegation: topLevelDelegation, Depth: 0})
+	}
+
+	for len(delegationsToSearch) > 0 {
+		currentDelegation := delegationsToSearch[0]
+		delegationsToSearch = delegationsToSearch[1:]
+
+		if state.HasTargetsRole(currentDelegation.Delegation.Name) {
+			// TODO: clarifying terminating / reachability
```

Comment:
terminating does not apply when we aren't querying for a specific namespace, so this can be taken out.

Reply:
Removed